packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -419,11 +419,42 @@ export class ProjectDataSource {
 
   async defaultSpecFileName (): Promise<string> {
     const { specPattern = [] } = await this.ctx.project.specPatterns()
+    let fileExtensionToUse: FileExtension = this.ctx.lifecycleManager.fileExtensionToUse
+
+    // If generating a component test then check whether there are JSX/TSX files present in the project.
+    // If project uses JSX then user likely wants to use JSX for their tests as well.
+    // JSX can be used (or not used) with a variety of frameworks depending on user preference/config, so
+    // the only reliable way to determine is whether there are files with JSX extension present
+    if (this.ctx.coreData.currentTestingType === 'component') {
+      debug('Checking for jsx/tsx files to determine file extension for default spec filename')
+      const projectJsxFiles = await this.ctx.file.getFilesByGlob(this.ctx.currentProject ?? '', '**/*.[jt]sx')
```

Comment:
Can this check not be framework based? If it's react -> `{fileExtensionToUse}x` else `${fileExtensionToUse}`?

Reply:
I was trying to think through some edge-cases, like a user having some `jsx` file for storybook but not wanting to have that pollute their generated Cypress tests if they don't plan on using jsx to write their CT tests. I think the proper solution is to have a more deterministic output like what framework they have and then, like @marktnoonan pointed out, having a user config override.

In the end this is just a suggestion for a filename and the user can change it if they want. I'm cool with this if this gets us closer, especially for react devs.